2016/proposals/result-aggregation-server-garg-prerit.md

```diff
@@ -0,0 +1,196 @@
+version: #34-Ubuntu SMP Tue Jan 8 21:47:00 UTC 2013 
+``` 
+Some or most of these information are being retrieved in the [installation-test-scripts](https://github.com/wking/swc-setup-installation-test). I would just need to collect them. 
+### Success/Failure of packages information collection 
+* **Successful Installs** list is already provided by the script. A list ```successes``` contains both checker and the version. All I need to do is grab the full name by ```checker.full_name()``` and ```version``` of the successfully installed packages. 
+* **Failed Checks** list is also provide by the script with the name of the package, error description and also the cause for some cases. e.g. : 
+``` 
+check for IPython script (ipython) failed: ( Name : IPython script (ipython) ) 
+  errors finding IPython script (ipython) version  <-- Description 
+causes: 
+check for IPython script (ipython) failed: 
+    could not find 'ipython' executable  <-- Cause 
+``` 
+###Other information 
+* Diagnose information 
+* Timestamp 
+* User email address ( optional ) 
+ 
+###Server and API 
+An API is to be developed to store the results generated by the test script. I shall be using Python's web framework [FLASK](http://flask.pocoo.org/) for this purpose. Reason being, flask is a [microframework](https://en.wikipedia.org/wiki/Microframework) with minimal complexity, and can be easily relied upon, as far as less complex projects are concerned. As we don't require a lot of endpoints or functionalities in this API, flask is the most suitable option as it seems. This would involve : 
+* Modification of the installation-test-scrips and giving the user an option to send the data to the server by making a [POST](https://en.wikipedia.org/wiki/POST_(HTTP)) request to the API endpoint with payload of all the information collected. Code Snippet : 
+``` 
+import requests 
+ 
+headers = { 
+    "Content-Type": "application/json", 
+} 
+HOST = "http://127.0.0.1:5000" 
+os_end_point = "/system_info/" 
+data = {"system_dist" : system_dist , "version": version} 
+uri = "%s%s" % (HOST, os_end_point) 
+response = requests.post(uri , data = json.dumps(data) , headers = headers) 
+``` 
+* This API will follow the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) standard, and all the data shall be send in [json](http://www.json.org/) to the API. 
+ 
+* Setting up the **Database**. As mentioned in the description of the idea, [SQLite](https://www.sqlite.org/) is preferred, but it suffers some problems in case 40-50 people send the data simultaneously. After going through the comparison given [here](https://www.digitalocean.com/community/tutorials/sqlite-vs-mysql-vs-postgresql-a-comparison-of-relational-database-management-systems), I personally prefer [MySQL](https://www.mysql.com/) in this project. But that can be decided later as per the discussion with mentors. 
```

Comment:
We're rather switch from SQLite to PostgreSQL, not MySQL. Why do you prefer MySQL over PostgreSQL?